src/http_server_accept_and_handle_conn.h

```diff
@@ -15,7 +15,10 @@
 extern "C" {
 #endif
 
-os_delta_ticks_t
+#define HTTP_SERVER_ACCEPT_TIMEOUT_MS (5)
```

Comment:
is 5 ms appropriate value? Sounds awfully fast

Reply:
Before trying to accept a new connection we need to lock the mutex. If we wait a long time in netconn_accept, then adv_post task will not be able to acquire the mutex and sending data to the server will be blocked.
If netconn_accept fails with timeout, then we sleep for 107 ms and during this time the mutex is not locked and adv_post can acquire it. The next time netconn_accept is called, it will accept any pending connections that clients try to establish during that sleep interval.
Unfortunately I could not find API like 'select' for the connections opened with 'netconn_new_with_callback'. 
Perhaps, it would be better to reduce HTTP_SERVER_ACCEPT_TIMEOUT_MS to 0, so that netconn_accept returns immediately if there is no pending incoming connections.

```
void
http_server_accept_and_handle_conn(struct netconn* const p_conn)
{
    struct netconn* p_new_conn = NULL;

    os_mutex_t p_mutex = g_p_mutex_accept_conn;
    if (NULL != p_mutex)
    {
        if (!os_mutex_try_lock(p_mutex))
        {
            LOG_DBG("Can't lock mutex, sleep for %u ms", HTTP_SERVER_ACCEPT_DELAY_MS);
            vTaskDelay(pdMS_TO_TICKS(HTTP_SERVER_ACCEPT_DELAY_MS));
            return;
        }
    }

    const err_t err = netconn_accept(p_conn, &p_new_conn);   // <-- here we're waiting for 5 ms
    // If there is no incoming connection, then we will sleep for HTTP_SERVER_ACCEPT_DELAY_MS (107 ms)

    if (ERR_OK != err)
    {
        if (NULL != p_mutex)
        {
            os_mutex_unlock(p_mutex);
        }
        if (ERR_TIMEOUT == err)
        {
            LOG_VERBOSE("netconn_accept ERR_TIMEOUT");
            vTaskDelay(pdMS_TO_TICKS(HTTP_SERVER_ACCEPT_DELAY_MS));  // <-- here is the main sleep
        }
        else if (ERR_ABRT == err)
        {
            LOG_ERR("netconn_accept ERR_ABRT");
        }
        else
        {
            LOG_ERR("netconn_accept: %d", err);
        }
        return;
    }
    
    // ...

```